scalyr_agent/otlp_client.py

```diff
@@ -0,0 +1,463 @@
+        return self.auth.authenticate()
+
+    def generate_status(self):
+        result = ScalyrClientSessionStatus()
+        result.total_requests_sent = self.total_requests_sent
+        result.total_requests_failed = self.total_requests_failed
+        result.total_request_bytes_sent = self.total_request_bytes_sent
+        result.total_compressed_request_bytes_sent = 0
+        result.total_response_bytes_received = 0
+        result.total_request_latency_secs = 0
+        result.total_connections_created = 0
+        result.total_compression_time = 0
+        return result
+
+    def close(self):
+        pass
+
+    def add_events_request(self, session_info=None, max_size=1 * 1024 * 1024 * 1024):
+        return OTLPAddEventsRequest()
+
+    def send(self, add_events_request, block_on_response=True):
+        try:
+            if self.__connection is None:
+                self.__connection = ConnectionFactory.connection(
+                    self.__full_address,
+                    self.__request_deadline,
+                    self.__ca_file,
+                    self.__intermediate_certs_file,
+                    self.headers,
+                    self.__use_requests,
+                    quiet=self.__quiet,
+                    proxies=self.__proxies,
+                )
+                self.total_connections_created += 1
+        except Exception as e:
+            error_code = (
+                    getattr(e, "error_code", "client/connectionFailed")
+                    or "client/connectionFailed"
+            )
+            return self.__wrap_response_if_necessary(
+                error_code, 0, "", block_on_response
+            )
+        self.ensure_auth_session()
+
+        self.__connection.post(OTLP_LOGS_PATH, self.__generate_body(add_events_request))
+        res = self.__connection.response()
+        return self.__wrap_response_if_necessary("success", 0, "success", block_on_response
+)
+
+    def augment_user_agent(self, fragments):
+        """Modifies User-Agent header (applies to all data sent to Scalyr)
+
+        @param fragments String fragments to append (in order) to the standard user agent data
+        @type fragments: List of six.text_type
+        """
+        self.headers["User-Agent"] = self.__get_user_agent(
+            SCALYR_VERSION, fragments
+        )
+
+    def __generate_body(self, add_events_request):
+        rl = ResourceLogs()
+        sl = ScopeLogs()
+        for event in add_events_request.events:
+            log_record = self.__event_to_log_record(event)
+            sl.log_records.append(log_record)
+        rl.scope_logs.append(sl)
+        return LogsData(resource_logs=[rl]).SerializeToString()
+
+    def __event_to_log_record(self, event):
+        return LogRecord(time_unix_nano=event.timestamp,
+                         body=PB2AnyValue(string_value=event.message),
+                         attributes=_encode_attributes(event.attrs))
+
+    def __get_user_agent(
```

Comment:
This whole method is copy pasted from scalyr_client.py.

Reply:
yes. This was intentional to avoid touching `scalyr_client.py`. I went ahead and moved this method and the wrap method to `util.py`